configure.ac

```diff
@@ -1611,6 +1611,36 @@ AC_SUBST(OPENTHREAD_ENABLE_SERVICE)
 AM_CONDITIONAL([OPENTHREAD_ENABLE_SERVICE], [test "${enable_service}" = "yes"])
 AC_DEFINE_UNQUOTED([OPENTHREAD_ENABLE_SERVICE],[${OPENTHREAD_ENABLE_SERVICE}],[Define to 1 if you want to enable Service])
 
+#
+# Settings in RAM
+#
+
+AC_ARG_ENABLE(settings-ram,
```

Comment:
Can we simply rely on defines, rather than a configuration option?

Couple reasons:
1. This feature should not be encouraged for general implementation and should only be used in very specific environments.
2. This feature does not really apply to the core OpenThread library.

Thoughts?

Reply:
Agree with this. The only difficulty is I'm not sure where's best to test and set `OPENTHREAD_SETTINGS_RAM` in the automake files, because a decision needs to be made when generating the Makefiles (for utils and qpg6095) about which source file to compile and which object file to link.

Reply:
How about including both `settings_ram.c` and `settings_flash.c` in the sources list and using `#if` feature flag in each source file? We generally prefer to use feature flags rather than automake source file selection - this reduces overhead for users of other build systems.